db/hints/manager.cc

```diff
@@ -450,25 +449,48 @@ future<> manager::change_host_filter(host_filter filter) {
     std::swap(_host_filter, filter);
     std::exception_ptr eptr = nullptr;
 
-    try {
-        // Iterate over existing hint directories and see if we can enable an endpoint manager
-        // for some of them
-        co_await lister::scan_dir(_hints_dir, lister::dir_entry_types::of<directory_entry_type::directory>(),
-                [this] (fs::path datadir, directory_entry de) {
-            const endpoint_id ep = endpoint_id{utils::UUID{de.name}};
-
-            const auto& topology = _proxy.get_token_metadata_ptr()->get_topology();
-            if (_ep_managers.contains(ep) || !_host_filter.can_hint_for(topology, ep)) {
-                return make_ready_future();
-            }
+    /* RAII lock for token metadata */ {
```

Comment:
Is this needed? The token metadata lock is already limited by the scope of `try { ... }`.